.github/workflows/main.yaml

```diff
@@ -0,0 +1,312 @@
+            DOCKER_FILE_PATH: Dockerfile.acapy # The docker path, file, is the relative path to the docker file from the root of the repo.
+            SOURCE_CONTEXT_DIR: docker/acapy   # The context dir, context, sets the context for the build. i.e. where the build will source files from 
+          - service: aries-endorser-db
+            GIT_REPO_URL: hyperledger/aries-endorser-service
+            PATH:  docker/wallet/config
+            BUILDER_IMAGE: 'quay.io/fedora/postgresql-13:13'
+          - service: aries-endorser-backup
+            GIT_REPO_URL: BCDevOps/backup-container
+            DOCKER_FILE_PATH: Dockerfile  # The docker path, file, is the relative path to the docker file from the root of the repo.
+            SOURCE_CONTEXT_DIR: docker    # The context dir, context, sets the context for the build. i.e. where the build will source files from
+          - service: aries-endorser-proxy
+            DOCKER_FILE_PATH: Dockerfile  # The docker path, file, is the relative path to the docker file from the root of the repo.
+            SOURCE_CONTEXT_DIR: proxy     # The context dir, context, sets the context for the build. i.e. where the build will source files from
+            SOURCE_IMAGE_REGISTRY: "artifacts.developer.gov.bc.ca/docker-remote/"
+            SOURCE_IMAGE_NAME: caddy
+            SOURCE_IMAGE_TAG: latest
+          - service: aries-endorser-api
+            GIT_REPO_URL: hyperledger/aries-endorser-service
+            DOCKER_FILE_PATH: Dockerfile.endorser # The docker path, file, is the relative path to the docker file from the root of the repo.
+            SOURCE_CONTEXT_DIR: endorser # The context dir, context, sets the context for the build. i.e. where the build will source files from
+            BASE_SOURCE_IMAGE_REGISTRY: artifacts.developer.gov.bc.ca/docker-remote/
+            BASE_SOURCE_IMAGE_NAME: python
+            BASE_SOURCE_IMAGE_TAG: 3.10-slim-buster
+            REGISTRY_USERNAME_SECRET_NAME: ARTIFACTORY_USERNAME 
+            REGISTRY_PASSWORD_SECRET_NAME: ARTIFACTORY_PASSWORD 
+    outputs:
+      aries-endorser-agent_digest: ${{ steps.digest.outputs.aries-endorser-agent_digest }} 
+      aries-endorser-backup_digest: ${{ steps.digest.outputs.aries-endorser-backup_digest }} 
+      aries-endorser-api_digest: ${{ steps.digest.outputs.aries-endorser-api_digest }}
+      aries-endorser-proxy_digest: ${{ steps.digest.outputs.aries-endorser-proxy_digest }}
+      aries-endorser-db_digest: ${{ steps.digests.outputs.aries-endorser-db_digest }}  
+
+    steps:
+    - name: Checkout
+      uses: actions/checkout@v4
+      with: 
+        repository: ${{ matrix.GIT_REPO_URL }}
+        ref: ${{ github.event.tag }} 
+      if: ${{ github.event.tag != '' }}
+
+    - name: Checkout
+      uses: actions/checkout@v4
+      with: 
+        repository: ${{ matrix.GIT_REPO_URL }}
+        ref: ${{ github.event.branch }} 
+      if: ${{ github.event.tag == '' }}
```

Comment:
It's unclear what is happening here.  Can you provide details on what's happening and why?

Reply:
If the tag is specified, it builds the image from tag. If the branch is specified instead of tag, it will build the image from branch. But if both tag and branch are specified , it will build image from main branch